src/main/java/com/example/demo/service/RecommendationService.java

```diff
@@ -0,0 +1,154 @@
+package com.example.demo.service;
+
+import com.example.demo.model.Song;
+import com.example.demo.repository.SongRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class RecommendationService {
+    
+    private static final double GENRE_PREFERENCE_THRESHOLD = 0.30;
+
+    @Autowired
+    private SongRepository songRepository;
+
+    private static final Logger logger = LoggerFactory.getLogger(RecommendationService.class);
+
+    public List<Song> getRecommendations(List<Song> inputSongs) {
+        // Analyze preferences from input songs
+        Map<String, Double> genrePreferences = analyzeGenrePreferences(inputSongs);
+        Set<String> preferredGenres = getPreferredGenres(genrePreferences);
+        
+        // Get frequently appearing artists
+        Set<String> frequentArtists = findFrequentArtists(inputSongs);
+        
+        // Add logging to debug
+        logger.info("Preferred genres: {}", preferredGenres);
+        logger.info("Frequent artists: {}", frequentArtists);
+
+        // Get all available songs from database
+        List<Song> availableSongs = songRepository.findAll();
```

Comment:
Spotify’s music library has grown to over 100 million tracks. load all songs to the database will cause performance issue.